providers/account.rb

```diff
@@ -210,3 +215,22 @@ def keygen_resource(exec_action)
     end
   end
 end
+
+def keypair_resource(exec_action)
+  new_resource.ssh_keypair.each do |name, key|
+    # avoid variable scoping issues in resource block
+    key_name, key_content = name, key
+
+    home = Etc.getpwnam(new_resource.username).dir
+    r = file "#{home}/.ssh/#{name}" do
+      content   key_content + "\n"
+      owner     new_resource.username
+      group     Etc.getpwnam(new_resource.username).gid
+      mode      '0600' unless key_name =~ /.pub$/
```

Comment:
I'm okay making this assumption for now. If it breaks in the future we can remove it.